From 6f65be43f733afaa6a5bc4bd5073adbb038fd565 Mon Sep 17 00:00:00 2001
From: spezi77 <spezi7713@gmx.net>
Date: Tue, 14 Dec 2021 17:18:13 +0100
Subject: [PATCH 5/5] base: ThemeOverlayApplier: Catch a potential NPE.

java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.om.OverlayIdentifier android.content.om.FabricatedOverlay.getIdentifier()' on a null object reference

Signed-off-by: spezi77 <spezi7713@gmx.net>
Change-Id: I5ce1bafcf5850dadcb53a997be49e03eecec1e53
---
 .../com/android/systemui/theme/ThemeOverlayApplier.java   | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayApplier.java b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayApplier.java
index e2d0bb9991ed..b41343216d5b 100644
--- a/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayApplier.java
+++ b/packages/SystemUI/src/com/android/systemui/theme/ThemeOverlayApplier.java
@@ -204,8 +204,12 @@ public class ThemeOverlayApplier implements Dumpable {
             HashSet<OverlayIdentifier> identifiersPending = new HashSet<>();
             if (pendingCreation != null) {
                 for (FabricatedOverlay overlay : pendingCreation) {
-                    identifiersPending.add(overlay.getIdentifier());
-                    transaction.registerFabricatedOverlay(overlay);
+                    try {
+                        identifiersPending.add(overlay.getIdentifier());
+                        transaction.registerFabricatedOverlay(overlay);
+                    } catch (NullPointerException e) {
+                        Log.e(TAG, "NPE for overlay.getIdentifier()", e);
+                    }
                 }
             }
 
-- 
2.25.1

